datafusion/functions/src/unicode/substr.rs

```diff
@@ -153,20 +175,15 @@ fn make_and_append_view(
     start: u32,
 ) {
     let substr_len = substr.len();
-    if substr_len == 0 {
-        null_builder.append_null();
-        views_buffer.push(0);
+    let sub_view = if substr_len > 12 {
+        let view = ByteView::from(*raw);
+        make_view(substr.as_bytes(), view.buffer_index, view.offset + start)
     } else {
-        let sub_view = if substr_len > 12 {
-            let view = ByteView::from(*raw);
-            make_view(substr.as_bytes(), view.buffer_index, view.offset + start)
-        } else {
-            // inline value does not need block id or offset
-            make_view(substr.as_bytes(), 0, 0)
-        };
-        views_buffer.push(sub_view);
-        null_builder.append_non_null();
-    }
+        // inline value does not need block id or offset
+        make_view(substr.as_bytes(), 0, 0)
```

Comment:
Seems the inline check here is duplicated with the one within `make_view`? 
I guess inline len(12) may be the internal logic in string view, and we should not expose and use it in datafusion?

Reply:
I would say the point we are checking size here is to help with the operation of "modifying the views directly". But I agree that maybe we could add more API upstream in arrow to hide the logic.

Reply:
> I would say the point we are checking size here is to help with the operation of "modifying the views directly". But I agree that maybe we could add more API upstream in arrow to hide the logic.

Maybe we can just let it run as same as the `> 12` branch? `ByteView::from(*raw)` seems cheap, maybe it is not more expansive than a `if else`? I am not sure.

Reply:
Oh I see what you mean. I think `ByteVIew` is not equal to inlined_view. The former one indicates the str whose size > 12, while the inlined str is represented by a simple `u128`. 
> Maybe we can just let it run as same as the > 12 branch?
It might actually work since `from<u128> for ByteView` would load meaningless `prefix, buffer_index, offset`, which does no harm. But I think it would be ambiguous. It's a good suggestion tho.